packages/core/src/lib/markdown-it/index.ts

```diff
@@ -144,14 +146,33 @@ markdownIt.renderer.rules.fence = (tokens: Token[],
   // wrap all lines with <span> so we can number them
   str = lines.map((line, index) => {
     const currentLineNumber = index + 1;
-    const rule = highlightRules.find(highlightRule =>
-      highlightRule.shouldApplyHighlight(currentLineNumber));
-    if (rule) {
-      return rule.applyHighlight(line, currentLineNumber);
+    // Rules that affects this line
+    const rules = highlightRules.filter(
+      highlightRule => highlightRule.shouldApplyHighlight(currentLineNumber));
+    if (rules.length === 0) {
+      // not highlighted
+      return `<span>${line}\n</span>`;
+    }
+    // FIXME: Reason why two rules cannot be applied on the same line
```

Comment:
Let's fix this before merging!

Reply:
U are referring to the fixme right, it is a stray comment